File: tests/test-minecraft/src/main/java/pize/tests/minecraft/game/GameRenderer.java
package pize.tests.minecraft.game;

import pize.graphics.util.batch.TextureBatch;

public class GameRenderer implements Renderer{

    private final Session session;

    private final TextureBatch batch;


    public GameRenderer(Session session){
        this.session = session;

        batch = new TextureBatch();
    }


    @Override
    public void render(){
        batch.begin();
        session.getScreenManager().render(batch);
        session.getIngameGui().render(batch);
        batch.end();
    }

    @Override
    public void resize(int width, int height){ }

    @Override
    public void dispose(){
        batch.dispose();
    }
    
}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/game/GameRenderer.kt
package pize.tests.minecraft.game

import pize.graphics.util.batch.TextureBatch

class GameRenderer(private val session: Session) : Renderer {
    private val batch: TextureBatch

    init {
        batch = TextureBatch()
    }

    override fun render() {
        batch.begin()
        session.screenManager.render(batch)
        session.ingameGui.render(batch)
        batch.end()
    }

    override fun resize(width: Int, height: Int) {}
    override fun dispose() {
        batch.dispose()
    }
}
